core/src/main/java/org/hisp/dhis/android/core/mockwebserver/ResponseController.java

```diff
@@ -0,0 +1,80 @@
+package org.hisp.dhis.android.core.mockwebserver;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static org.apache.commons.lang3.StringUtils.EMPTY;
+
+public class ResponseController {
+
+    public static final String GET = "GET";
+    public static final String POST = "POST";
+    public static final String PUT = "PUT";
+    public static final String DELETE = "DELETE";
+
+    public static final String API_ME_PATH = "/api/me?.*";
+    public static final String API_SYSTEM_INFO_PATH = "/api/system/info?.*";
+
+    private Map<String, LinkedHashMap<String, String>> methodsMap;
+    private Map<String, Integer> codeResponses;
+
+    ResponseController(){
+        initMaps();
+    }
+
+    private void initMaps() {
+        codeResponses = new HashMap<>();
+        methodsMap = new HashMap<>();
+        methodsMap.put(GET, new LinkedHashMap<>());
+        methodsMap.put(POST, new LinkedHashMap<>());
+        methodsMap.put(PUT, new LinkedHashMap<>());
+        methodsMap.put(DELETE, new LinkedHashMap<>());
+    }
+
+    void populateInternalResponses(){
+
```

Comment:
The idea is to init here methodsMap with the SDK json/paths. If so, I think we could share the new dhis2Dispatcher